File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Diagrams/addcommenttodiagram.rb
module Asposediagramjava
  module AddCommentToDiagram
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        # Add comment
        diagram.getPages().getPage(0).addComment(7.205905511811023, 3.880708661417323, "test@")

        # Save as VDX
        diagram.save(data_dir + "AddComment.vdx", Rjb::import('com.aspose.diagram.SaveFileFormat').VDX)

        puts "Added comment successfully!"
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithDiagrams/AddCommentToDiagram.php
<?php
namespace Aspose\Diagram\WorkingwithDiagrams;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class AddCommentToDiagram{

    public static function run($dataDir=null){
        # Create instance of Diagram
        $diagram = new Diagram($dataDir."Drawing.vsd");

        # Add comment
        $diagram->getPages()->getPage(0)->addComment(7.205905511811023, 3.880708661417323, "test@");

        # Save as VDX
        $saveFileFormat=new SaveFileFormat();
        $diagram->save($dataDir."AddComment.vdx", $saveFileFormat->VDX);
        print "Added comment successfully!".PHP_EOL;
    }

}